torch/csrc/jit/tensorexpr/loopnest.h

```diff
@@ -1,17 +1,25 @@
 #pragma once
 
-#include <memory>
+#include <vector>
 #include <unordered_map>
+#include <unordered_set>
 
-#include <c10/util/Logging.h>
-#include <torch/csrc/jit/tensorexpr/expr.h>
-#include <torch/csrc/jit/tensorexpr/ir.h>
-#include <torch/csrc/jit/tensorexpr/tensor.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
 
 namespace torch {
 namespace jit {
 namespace tensorexpr {
 
+class Expr;
+class Var;
+class Tensor;
```

Comment:
Minor nit: In general, people might be surprised that although "loopnest.h" uses "Tensor" as part of its interface, it didn't include "tensor.h". Therefore, they have to include "tensor.h" themselves. Not a deal breaker, but a bit surprising to most people. In general, it's good to include classes in your public method, and we can forward declare on private methods and members.